lib/KImg/index.vue

```diff
@@ -356,20 +362,33 @@
     },
     created() {
       if (!this.isDecorative && !this.altText) {
-        throw new Error('Missing required prop - provide altText or indicate isDecorative');
+        this.onError(new Error('Missing required prop - provide altText or indicate isDecorative'));
       }
     },
     methods: {
+      /**
+       * Emitted when => with:
+       * - the image fails to load => the DOM event that triggered the error
+       * - there is no alt text => error obj
+       * - the aspect ratio is invalid => error obj
+       *
+       * @param {UIEvent|Error} event
+       */
       onError(event) {
-        /**
-         * Emitted when the image fails to load. The DOM event that triggered the error is available in the payload.
-         */
-        this.$emit('error', event);
+        if (this.$listeners && this.$listeners.error) {
```

Comment:
Oh well, I said it more because normally an onError handler in <img> tags is usually more for handling image loading errors, and it seemed appropriate to me that a consumer expects a KImg onError to be quite similar to an <img> onError, but, I guess doing validations doesn't hurt either.

Reply:
You're right, it is a little awkward in that respect. The sense that I got is that we wouldn't normally handle the other errors and they shouldn't arise in 'production', since those seem most useful within development. So the end result means you should structure your listener to re-raise anything it doesn't expect, which isn't too dissimilar from a typical catch

Reply:
^^ And a positive from this is that it promotes writing more explicit code